processors/enduserprocessor/enduserprocessor.go

```diff
@@ -0,0 +1,534 @@
+package enduserprocessor
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/PaesslerAG/jsonpath"
+	"github.com/dgrijalva/jwt-go"
+	jsoniter "github.com/json-iterator/go"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.uber.org/zap"
+
+	"github.com/hypertrace/collector/processors/enduserprocessor/hash"
+	"github.com/hypertrace/collector/processors/piifilterprocessor"
+)
+
+const (
+	enduserIDAttribute      = "enduser.id"
+	enduserRoleAttribute    = "enduser.role"
+	enduserScopeAttribute   = "enduser.scope"
+	enduserSessionAttribute = "session.id"
+)
+
+type config struct {
+	EndUser
+	hashAlgorithm hash.Algorithm
+}
+
+type processor struct {
+	logger             *zap.Logger
+	attributeConfigMap map[string][]config
+}
+
+type user struct {
+	id      string
+	role    string
+	scope   string
+	session string
+}
+
+var _ processorhelper.TProcessor = (*processor)(nil)
+
+func (p *processor) ProcessTraces(_ context.Context, traces pdata.Traces) (pdata.Traces, error) {
+	rss := traces.ResourceSpans()
+	for i := 0; i < rss.Len(); i++ {
+		rs := rss.At(i)
+
+		ilss := rs.InstrumentationLibrarySpans()
+		for j := 0; j < ilss.Len(); j++ {
+			ils := ilss.At(j)
+
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+
+				span.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+					unindexedKey := piifilterprocessor.UnindexedKey(k)
```

Comment:
I know this is ported from the previous collector but let's avoid leaking this kind of logic. The code behind this function is simple enough to copy into this processor, also it is not exclusive for PII filter.